### PR TITLE
convert summary metrics to cumulatives and gauges.

### LINKED ISSUE #151: Add support to convert summary metrics
SD currently doesn't support Summary metrics. In order to support summary metrics from sources like Prometheus it is desirable to convert summary metrics to following as done for [Java SD exporter.](https://github.com/census-instrumentation/opencensus-java/issues/1514)
Summary.Sum to CUMULATIVE_DOUBLE
Summary.Count to CUMULATIVE_INT64
Summary.Percentiles to CUMULATIVE_GAUGE for the value and percentile as value for Label "percentile"

### PR DESCRIPTION
**Pull request by rghetia**
- convert summary.sum to cumulative_int64
- convert summary.count to cumulative_double
- convert summary.percentile to gauge with label "percentile"

fixes #151

### CODE REVIEW

File: metrics_proto.go [hunk 1]
```diff
@@ -45,12 +45,17 @@ import (
...
+	Key:         "percentile",
+	Description: "the value at a given percentile of a distribution",
+}
 
 type metricProtoPayload struct {
 	node             *commonpb.Node
 	resource         *resourcepb.Resource
 	metric           *metricspb.Metric
 	additionalLabels map[string]labelValue
+	convertedMetrics []*metricspb.Metric // Summary metrics.
```
Comment (songy23) [inline metrics_proto.go]: Is it possible to combine this field and `metric` field above? e.g use one `metrics []*metricspb.Metric` field in `metricProtoPayload`.

File: metrics_proto.go [hunk 2]
```diff
@@ -91,20 +207,41 @@ func (se *statsExporter) handleMetricsProtoUpload(payloads []*metricProtoPayload
 
 	for _, payload := range payloads {
 		// Now create the metric descriptor remotely.
-		if err := se.createMetricDescriptor(ctx, payload.metric, payload.additionalLabels); err != nil {
-			span.SetStatus(trace.Status{Code: 2, Message: err.Error()})
-			return err
+		if payload.metric.GetMetricDescriptor().GetType() == metricspb.MetricDescriptor_SUMMARY {
+			payload.convertedMetrics = se.convertSummaryMetrics(payload.metric)
```
Comment (songy23) [inline metrics_proto.go]: Consider converting summary metrics at `ExportMetricsProto`: https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/pull/152/files#diff-c4657249ba7aeb2b6dc05cd7ec12497eR74, then you may be able to get rid of the conditional checks here and below.

File: metrics_proto.go (hunk 1 above)
Comment (rghetia) [reply metrics_proto.go]: should be. Let me give it a shot.

File: metrics_proto.go (hunk 2 above)
Comment (rghetia) [reply metrics_proto.go]: good point. Let me try.

File: metrics_proto.go [hunk 3]
```diff
@@ -53,9 +53,8 @@ var percentileLabelKey = &metricspb.LabelKey{
 type metricProtoPayload struct {
 	node             *commonpb.Node
 	resource         *resourcepb.Resource
-	metric           *metricspb.Metric
+	metric           []*metricspb.Metric
```
Comment (songy23) [inline metrics_proto.go]: Nit: would it be better to keep one `metric` per `payload` and have multiple `payload`s for `Summary` type?

/cc @odeke-em any reason we chose one `metric` per `payload` at first?

Comment (songy23) [reply metrics_proto.go]: Also consider renaming this to `metrics` if we keep it as a slice.

**Pull request merged**